keps/sig-network/4427-relaxed-dns-search-validation/README.md

```diff
@@ -0,0 +1,813 @@
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Currently, Kubernetes applies the same restrictions to DNS search strings as it does to object names.
+These restrictions are derived from [RFC-1123](https://datatracker.ietf.org/doc/html/rfc1123). However,
+DNS name segments may contain more characters than are allowed in host names (see [RFC-2181](https://datatracker.ietf.org/doc/html/rfc2181#section-11)).
+In practice, DNS names sometimes contain characters, such as underscores, which would not be allowed by RFC-1123.
+
+This KEP proposes relaxing the checks on DNS search strings, specifically, without changing the restrictions on Kubernetes object
+names or other uses of DNS names within Kubernetes. Allowing these values in the `searches` field of `dnsConfig` allows pods to
```

Comment:
In the case of a loadBalancer, the hostname is always an internet host name, though. Right? Search strings are frequently used to form host names, but not always. In our internal case that motivated this proposal, these search strings are actually used to look up SRV records - so neither the full names nor the search strings name a host.

Underscores are actually required in the protocol and service portion of the record. 

Would this proposal make more sense if it omitted the comments about object names and focused on search strings used for names that are not internet hostnames? (with the side-effect that the check is relaxed for all search strings since they can't be distinguished)

Reply:
I'll rewrite the motivation to focus on that case.
The example SRV record would be `_sip._tcp.abc_d.example.com` and the search string would be `abc_d.example.com` (which has no A record associated with it and does not correspond to any host).